pkg/ipam/crd.go

```diff
@@ -549,14 +549,34 @@ func (n *nodeStore) isIPInReleaseHandshake(ip string) bool {
 }
 
 // allocateNext allocates the next available IP or returns an error
-func (n *nodeStore) allocateNext(allocated ipamTypes.AllocationMap, family Family) (net.IP, *ipamTypes.AllocationIP, error) {
+func (n *nodeStore) allocateNext(allocated ipamTypes.AllocationMap, family Family, owner string) (net.IP, *ipamTypes.AllocationIP, error) {
 	n.mutex.RLock()
 	defer n.mutex.RUnlock()
 
 	if n.ownNode == nil {
 		return nil, nil, fmt.Errorf("CiliumNode for own node is not available")
 	}
 
+	// Check if IP has a custom owner (only supported in manual CRD mode)
+	if n.conf.IPAMMode() == ipamOption.IPAMCRD && len(owner) != 0 {
+		for ip, ipInfo := range n.ownNode.Spec.IPAM.Pool {
+			if ipInfo.Owner == owner {
+				parsedIP := net.ParseIP(ip)
+				if parsedIP == nil {
+					log.WithFields(logrus.Fields{
+						fieldName: n.ownNode.Name,
+						"ip":      ip,
+					}).Warning("Unable to parse IP in CiliumNode custom resource")
+					return nil, nil, fmt.Errorf("invalid custom ip %s for %s. ", ip, owner)
```

Comment:
One last question: Should we really make this a hard error and refuse to assign an IP to the pod? It might be better to allow a generic IP to be allocated. What do you think @llhhbc ?

Reply:
I may not be sure about other scenarios, but in the scenario we use, if the IP I specified is not successfully assigned and another IP is randomly assigned, it may cause serious business problems. Currently we are using it in database scenarios, some databases have strict control over IP

Reply:
Thank you for the clarification. Given that use case I think it's reasonable to block pod creation until the error is fixed. Thanks!